react/features/analytics/AnalyticsEvents.js

```diff
@@ -1,663 +1,474 @@
+ * video.
  *
  * @type {String}
  */
-export const SHARED_VIDEO_STOPPED = 'sharedvideo.stoped';
+export const ACTION_UNPINNED = 'unpinned';
 
 /**
- * The shared YouTube video had its volume change.
+ * The identifier for the "pressed" action for shortcut events. This action
+ * means that a button was pressed (and not yet released).
  *
  * @type {String}
  */
-export const SHARED_VIDEO_VOLUME_CHANGED = 'sharedvideo.volumechanged';
+export const ACTION_SHORTCUT_PRESSED = 'pressed';
 
 /**
- * Pressed the keyboard shortcut for toggling audio mute.
+ * The identifier for the "released" action for shortcut events. This action
+ * means that a button which was previously pressed was released.
  *
  * @type {String}
  */
-export const SHORTCUT_AUDIO_MUTE_TOGGLED = 'shortcut.audiomute.toggled';
+export const ACTION_SHORTCUT_RELEASED = 'released';
 
 /**
- * Pressed the keyboard shortcut for toggling chat panel display.
+ * The identifier for the "triggered" action for shortcut events. This action
+ * means that a button was pressed, and we don't care about whether it was
+ * released or will be released in the future.
  *
  * @type {String}
  */
-export const SHORTCUT_CHAT_TOGGLED = 'shortcut.chat.toggled';
+export const ACTION_SHORTCUT_TRIGGERED = 'triggered';
 
 /**
- * Toggled the display of the keyboard shortcuts help dialog.
- *
- * @type {String}
+ * The name of the keyboard shortcut or toolbar button for muting audio.
  */
-export const SHORTCUT_HELP = 'shortcut.shortcut.help';
+export const AUDIO_MUTE = 'audio.mute';
 
 /**
- * Pressed the keyboard shortcut for togglgin raise hand status.
- *
- * @type {String}
+ * The name of the keyboard shortcut or toolbar button for muting video.
  */
-export const SHORTCUT_RAISE_HAND_CLICKED = 'shortcut.raisehand.clicked';
+export const VIDEO_MUTE = 'video.mute';
 
 /**
- * Pressed the keyboard shortcut for toggling screenshare.
+ * Creates an event which indicates that a certain action was requested through
+ * the jitsi-meet API.
  *
- * @type {String}
+ * @param {Object} action - The action which was requested through the
+ * jitsi-meet API.
+ * @param {Object} attributes - Attributes to attach to the event.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
  */
-export const SHORTCUT_SCREEN_TOGGLED = 'shortcut.screen.toggled';
+export const createApiEvent = function(action, attributes = {}) {
+    return {
+        action,
+        attributes,
+        source: 'jitsi-meet-api'
+    };
+};
 
 /**
- * Toggled the display of the speaker stats dialog.
+ * Creates an event which indicates that the audio-only mode has been turned
+ * off.
  *
- * @type {String}
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
  */
-export const SHORTCUT_SPEAKER_STATS_CLICKED = 'shortcut.speakerStats.clicked';
+export const createAudioOnlyDisableEvent = function() {
+    return {
+        name: 'audio.only.disabled'
```

Comment:
Some of these events are name only because all other fields will get filled in automatically? It's not clear to me immediately if that's the case. What's the difference between action and name again? One is used for TYPE_OPERATIONAL and the other for TYPE_UI/TYPE_TRACK? If that's the case, I wonder if a future todo could be to hide that detail so a consistent key is used, but that might also be too much magic.

Reply:
In the lower layer (the analytics handler), "name" doesn't exist (except for PAGE events, which ONLY have a name). In the jitsi-meet/lib-jitsi-meet we use "name" as a shortcut, and it it used to fill in any of the required fields which are empty. So it's a lazy way to call the API (although I guess equally lazy as "action: '...'")

Reply:
I guess I can get rid of this usage of "name"